vyper/semantics/analysis/utils.py

```diff
@@ -96,6 +96,16 @@ def get_expr_info(self, node: vy_ast.VyperNode) -> ExprInfo:
 
             return ExprInfo(t, location=location, is_constant=is_constant)
 
+        # If it's a Subscript, propagate the subscriptable varinfo
+        if isinstance(node, vy_ast.Subscript):
+            varinfo = self.get_expr_info(node.value)
+            return ExprInfo(
+                t,
```

Comment:
Maybe ExprInfo.from_varinfo here

Reply:
This would not work for nested arrays because we want the outermost type with the innermost varinfo.

```
@external
def test(a: uint256[4][4]):
    a[0][1] = 1
```
In this example, for `a[0][1]`, we want the type `uint256` with the varinfo of `a`.

Reply:
can also use `ExprInfo.copy_with_type`!

Reply:
oops thanks! it was right below `from_varinfo` :sweat_smile: